cucumber/authenticators_azure/features/support/authn_azure_helper.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+# Utility methods for Azure authenticator
+
+module AuthnAzureHelper
+  include AuthenticatorHelpers
+
+  def set_azure_provider_uri_variable(value = azure_provider_uri)
+    set_azure_variable("provider-uri", value)
+  end
+
+  def retrieve_azure_access_token
+    command = "curl 'http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fmanagement.azure.com%2F' -H Metadata:true -s | jq -r '.access_token'"
+    @azure_token = run_command_in_machine(azure_machine_ip, azure_machine_username, azure_machine_password, command)
+  rescue Exception => err
+    raise "Failed to fetch azure token with reason: #{err}"
+  end
+
+  def authenticate_azure_token(service_id:, account:, username:, azure_token: parsed_azure_token)
+    username = username.gsub("/", "%2F")
+    path = "#{conjur_hostname}/authn-azure/#{service_id}/#{account}/#{username}/authenticate"
+
+    payload = {}
+    unless azure_token.nil?
+      payload["jwt"] = azure_token
+    end
+
+    post(path, payload)
```

Comment:
this triggers a POST to conjur with path `/authenticate` the jwt in the body (payload)? This mocks the actual request?

Reply:
1. We don't mock anything in integration tests.
2. Yes, it does exactly what you said in the first sentence.